pkg/client/cache/imagestream.go

```diff
@@ -0,0 +1,92 @@
+package cache
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	"k8s.io/kubernetes/pkg/labels"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+// StoreToImageStreamLister gives a store List and Exists methods. The store must contain only image streams.
+type StoreToImageStreamLister struct {
+	cache.Indexer
+}
+
+// Exists checks if the given image stream exists in the store.
+func (s *StoreToImageStreamLister) Exists(stream *imageapi.ImageStream) (bool, error) {
```

Comment:
I am following the upstream pattern. I would guess they have it for a reason there.